### ISSUE TITLE
construct-constructor-property: optional `scope?: Construct` parameter is reported as not a Construct type

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#603]
### Describe the bug

`construct-constructor-property` checks that the first constructor parameter is typed `Construct`. When the parameter is optional (`scope?: Construct`) or typed `Construct | undefined`, the resolved type is a union with `undefined`; the base-chain walk rejects unions, so the rule reports `invalidConstructorType` even though the declared type is `Construct`.

This is exactly the signature of `Stack` itself (`constructor(scope?: Construct, id?: string, props?: StackProps)`), so a Stack subclass that mirrors its base signature is reported. Stack subclasses have been in scope for this rule since #570.

`src/my-stack.ts`:

```ts
import { Stack, StackProps } from "aws-cdk-lib";
import { Construct } from "constructs";

export class MyStack extends Stack {
  constructor(scope?: Construct, id?: string, props?: StackProps) {
    super(scope, id, props);
  }
}
```

Rule enabled: `awscdk/construct-constructor-property: "error"`.

```
$ npx eslint src/my-stack.ts
  5:15  error  Constructor of a class extending Construct must have the type 'Construct' for the first parameter  awscdk/construct-constructor-property

$ npx oxlint --format=unix -c .oxlintrc.json src/my-stack.ts
src/my-stack.ts:5:15: Constructor of a class extending Construct must have the type 'Construct' for the first parameter [Error/awscdk(construct-constructor-property)]
```

`constructor(scope: Construct | undefined, id: string)` on a Construct subclass is reported the same way.

### Expected Behavior

No report: the parameter is named `scope` and its declared type is `Construct`; optionality does not change that.

### Current Behavior

`invalidConstructorType` is reported by both plugins.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.2 / oxlint-plugin-awscdk 1.0.2

### Language Version

TypeScript 6.0.3 (eslint) / TypeScript 7.0.2 via oxlint-tsgolint 7.0.2001 (oxlint)

### Additional Context

```yaml
# machine-readable summary
rules: ["construct-constructor-property"]
symptom: ["false-positive"]
affected_packages: ["eslint-plugin-awscdk", "oxlint-plugin-awscdk"]
root_cause: "packages/eslint/src/rules/construct-constructor-property.ts:106 -> core/ts-type/checker/is-extends-target-super-class.ts:15 (`Construct | undefined` has no symbol); packages/oxlint/src/rules/construct-constructor-property.ts:103 -> core/ts-type/checker/is-extends-target-super-class.ts:21 (union bail-out)"
verified_on: "main @ 9caea55 (eslint-plugin-awscdk 4.4.2, oxlint-plugin-awscdk 1.0.2)"
related_issues: ["#570", "#574", "#567"]
```

Same shape as #574 / #567: the type handed to `isConstructType` is `Construct | undefined`. Verified fix direction (eslint): stripping the nullish members with the existing `findNonNullableType` before the check, as `prefer-grants-property` and `no-unused-props` already do, makes the fixture pass with `examples/` still at 43 errors. The oxlint finder is mirrored, so the same change applies there.

Reproduction environment:

- macOS 14.1, Node.js v24.18.0, pnpm 10.29.3
- aws-cdk-lib 2.268.0 / constructs 10.8.1
- ESLint 10.9.1 + typescript-eslint 8.69.0 + TypeScript 6.0.3
- oxlint 1.81.0 + oxlint-tsgolint 7.0.2001 (corsa-oxlint 1.13.1)
